homeassistant/components/deconz/light.py

```diff
@@ -158,111 +170,139 @@ def color_mode(self) -> str:
         return color_mode
 
     @property
-    def brightness(self):
+    def brightness(self) -> int | None:
         """Return the brightness of this light between 0..255."""
-        return self._device.brightness
+        if self._device.brightness:
+            return int(self._device.brightness)
+        return None
 
     @property
-    def color_temp(self):
+    def color_temp(self) -> int:
         """Return the CT color value."""
-        return self._device.color_temp
+        return int(self._device.color_temp)
 
     @property
-    def hs_color(self) -> tuple:
+    def hs_color(self) -> tuple[float, float]:
         """Return the hs color value."""
-        return (self._device.hue / 65535 * 360, self._device.saturation / 255 * 100)
+        return (
+            int(self._device.hue) / 65535 * 360,
+            int(self._device.saturation) / 255 * 100,
+        )
 
     @property
-    def xy_color(self) -> tuple | None:
+    def xy_color(self) -> tuple[float, float] | None:
         """Return the XY color value."""
-        return self._device.xy
+        if self._device.xy is not None:
+            x_value: float = self._device.xy[0]
+            y_value: float = self._device.xy[1]
```

Comment:
I would expected `xy` to be a tuple of float already? Is this an upstream issue?